tools/depmod.c

```diff
@@ -1351,53 +1348,51 @@ static int depmod_modules_search_dir(struct depmod *depmod, DIR *d, size_t basel
 				close(fd);
 				continue;
 			}
-			path[baselen + namelen] = '/';
-			path[baselen + namelen + 1] = '\0';
+
+			strbuf_pushchar(path, '/');
 			err = depmod_modules_search_dir(depmod, subdir,
-							baselen + namelen + 1, s_path);
+							baselen + namelen + 1, path);
+			strbuf_popchar(path);
 			closedir(subdir);
 		} else {
-			err = depmod_modules_search_file(depmod, baselen, namelen, path);
+			err = depmod_modules_search_file(depmod, baselen, namelen,
+							 strbuf_str(path));
 		}
 
 		if (err < 0) {
-			path[baselen + namelen] = '\0';
-			ERR("failed %s: %s\n", path, strerror(-err));
+			ERR("failed %s: %s\n", strbuf_str(path), strerror(-err));
 			err = 0; /* ignore errors */
 		}
+
+		strbuf_popchars(path, namelen);
 	}
 	return err;
 }
 
 static int depmod_modules_search_path(struct depmod *depmod, const char *path)
 {
-	char buf[256];
-	_cleanup_(scratchbuf_release) struct scratchbuf s_path_buf =
-		SCRATCHBUF_INITIALIZER(buf);
-	char *path_buf;
+	_cleanup_strbuf_ struct strbuf s_path_buf;
 	DIR *d;
 	size_t baselen;
 	int err;
 
+	strbuf_init_with_stack(&s_path_buf, 256);
+
 	d = opendir(path);
 	if (d == NULL) {
 		err = -errno;
 		ERR("could not open directory %s: %m\n", path);
 		return err;
 	}
 
-	baselen = strlen(path);
-
-	if (scratchbuf_alloc(&s_path_buf, baselen + 2) < 0) {
+	baselen = strlen(path) + 1;
+	if (!strbuf_reserve(&s_path_buf, baselen)) {
 		err = -ENOMEM;
 		goto out;
 	}
-	path_buf = scratchbuf_str(&s_path_buf);
 
-	memcpy(path_buf, path, baselen);
-	path_buf[baselen] = '/';
-	baselen++;
-	path_buf[baselen] = '\0';
+	strbuf_pushmem(&s_path_buf, path, baselen - 1);
```

Comment:
`baselen - 1` is `strlen(path) + 1 - 1` is `strlen(path)` so we can use `strbuf_pushchars` here as well.